src/content/developers/tutorials/erc20-annotated-code/index.md

```diff
@@ -0,0 +1,907 @@
+created.
+
+```solidity
+    /**
+     * @dev Sets `amount` as the allowance of `spender` over the `owner` s tokens.
+     *
+     * This internal function is equivalent to `approve`, and can be used to
+     * e.g. set automatic allowances for certain subsystems, etc.
+     *
+     * Emits an {Approval} event.
+     *
+     * Requirements:
+     *
+     * - `owner` cannot be the zero address.
+     * - `spender` cannot be the zero address.
+     */
+    function _approve(address owner, address spender, uint256 amount) internal virtual {
+        require(owner != address(0), "ERC20: approve from the zero address");
+        require(spender != address(0), "ERC20: approve to the zero address");
+
+        _allowances[owner][spender] = amount;
+```
+
+&nbsp;
+
+Emit an `Approval` event. Depending on how the application is written, the spender contract can be told about the
+approval either by the owner or by a server that listens to these events.
+
+```solidity
+        emit Approval(owner, spender, amount);
+    }
+
+```
+
+### Modify The Decimals Variable  {#modify-the-decimals-variable}
+
+This function modifies the `_decimals` variable which is used to tell user interfaces how to interpret the amount. 
+You should call it from the constructor. It would be dishonest to call it at any subsequent point, and applications
+are not designed to handle it.
+
+```solidity
+
+
+    /**
+     * @dev Sets {decimals} to a value other than the default one of 18.
+     *
+     * WARNING: This function should only be called from the constructor. Most
+     * applications that interact with token contracts will not expect
+     * {decimals} to ever change, and may work incorrectly if it does.
+     */
+    function _setupDecimals(uint8 decimals_) internal {
+        _decimals = decimals_;
+    }
+```
+
+### Hooks  {#hooks}
+
+This is the hook function to be called during transfers. It is empty here, buf if you need
+to do something you just override it.
+
+```solidity
+
+    /**
+     * @dev Hook that is called before any transfer of tokens. This includes
+     * minting and burning.
+     *
+     * Calling conditions:
+     *
+     * - when `from` and `to` are both non-zero, `amount` of ``from``'s tokens
+     * will be to transferred to `to`.
+     * - when `from` is zero, `amount` tokens will be minted for `to`.
+     * - when `to` is zero, `amount` of ``from``'s tokens will be burned.
+     * - `from` and `to` are never both zero.
+     *
+     * To learn more about hooks, head to xref:ROOT:extending-contracts.adoc#using-hooks[Using Hooks].
+     */
+    function _beforeTokenTransfer(address from, address to, uint256 amount) internal virtual { }
+}
+```
+
+# Conclusion {#conclusion}
+
+By this point you should understand the OpenWhisk ERC20 code and why it is written like that. Hopefully this knowledge will help you develop your own contracts and applications.
```

Comment:
Would consider re-wording if this is meant as a "conclusion"... 
- Maybe review/highlight a few benefits of using OpenWhisk
- Instead of using "Hopefully" maybe change into a call-to-action, like "Now use what you've learned to develop your own contracts and applications!"